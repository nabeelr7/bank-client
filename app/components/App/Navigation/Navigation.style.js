import styled from 'styled-components';
import { Menu } from 'antd';

export const StyledMenuItem = styled(Menu.Item)`
  &&& {
    margin-top: 0;
  }
`;
