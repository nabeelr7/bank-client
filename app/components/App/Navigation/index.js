import React from 'react';
import { Menu } from 'antd';
import { routes } from 'utils';

import {
  DesktopOutlined,
  WalletOutlined,
  HistoryOutlined,
  SettingOutlined,
  CreditCardOutlined,
  LineChartOutlined,
  BankOutlined,
} from '@ant-design/icons';
import { makeSelectLocation } from 'containers/App/selectors';
import { NavLink } from 'react-router-dom/cjs/react-router-dom.min';
import { useSelector } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { StyledMenuItem } from './Navigation.style';

const items = [
  {
    id: 1,
    name: routes.dashboard.name,
    path: routes.dashboard.path,
    icon: <DesktopOutlined />,
    disabled: false,
  },
  {
    id: 2,
    name: routes.payment.name,
    path: routes.payment.path,
    icon: <WalletOutlined />,
    disabled: false,
  },
  {
    id: 3,
    name: 'History',
    path: '3',
    icon: <HistoryOutlined />,
    disabled: true,
  },
  {
    id: 4,
    name: 'Cards',
    path: '4',
    icon: <CreditCardOutlined />,
    disabled: true,
  },
  {
    id: 5,
    name: 'Credits',
    path: '5',
    icon: <LineChartOutlined />,
    disabled: true,
  },
  {
    id: 6,
    name: 'Deposits',
    path: '6',
    icon: <BankOutlined />,
    disabled: true,
  },
  {
    id: 7,
    name: 'Settings',
    path: '7',
    icon: <SettingOutlined />,
    disabled: true,
  },
];

const stateSelector = createStructuredSelector({
  location: makeSelectLocation(),
});

export default function Navigation() {
  const {
    location: { pathname },
  } = useSelector(stateSelector);

  return (
    <Menu theme="light" mode="inline" selectedKeys={[pathname]}>
      {items.map((item) => (
        <StyledMenuItem
          key={item.path}
          icon={item.icon}
          disabled={item.disabled}
        >
          <NavLink to={item.path}>{item.name}</NavLink>
        </StyledMenuItem>
      ))}
    </Menu>
  );
}
