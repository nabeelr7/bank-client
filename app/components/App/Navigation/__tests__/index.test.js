import React from 'react';
import { render } from '@testing-library/react';
import Navigation from '../index';

describe('<Navigation />', () => {
  it('should match the snapshot', () => {
    const { container } = render(<Navigation />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
