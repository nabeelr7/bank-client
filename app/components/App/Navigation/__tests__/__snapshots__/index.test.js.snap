// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> should match the snapshot 1`] = `
<ul
  class="ant-menu ant-menu-light ant-menu-root ant-menu-inline"
  role="menu"
>
  <li
    class="ant-menu-item Navigationstyle__StyledMenuItem-wt3miw-0 dUoMzc ant-menu-item-selected"
    role="menuitem"
    style="padding-left: 24px;"
  >
    <span
      aria-label="desktop"
      class="anticon anticon-desktop Navigationstyle__StyledDesktopOutlined-wt3miw-1 emSucX"
      role="img"
    >
      <svg
        aria-hidden="true"
        class=""
        data-icon="desktop"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M928 140H96c-17.7 0-32 14.3-32 32v496c0 17.7 14.3 32 32 32h380v112H304c-8.8 0-16 7.2-16 16v48c0 4.4 3.6 8 8 8h432c4.4 0 8-3.6 8-8v-48c0-8.8-7.2-16-16-16H548V700h380c17.7 0 32-14.3 32-32V172c0-17.7-14.3-32-32-32zm-40 488H136V212h752v416z"
        />
      </svg>
    </span>
    <span>
      <span>
        Dashboard
      </span>
    </span>
  </li>
</ul>
`;
